app/models/user.rb

```diff
@@ -1,16 +1,39 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+  has_many :rents, dependent: :destroy
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, omniauth_providers: [:google_oauth2]
   # Hooks
   before_validation :generate_verification_code, on: :create
 
   def generate_verification_code
     self.verification_code = AuthenticableEntity.verification_code
   end
 
-  has_many :rents, dependent: :destroy
+  def self.create_from_omniauth(params)
+    user = find_or_create_by(provider: params.provider, uid: params.uid)
+    user.update(User.omniauth_credentials_params(params.credentials))
+    user.update(User.omniauth_info_params(params.info))
+    user.update(password: Devise.friendly_token[0, 20])
```

Comment:
take care here, you are doing 3 update querys, maybe you can keep the omniauth example and ignore the abc offence in pos of performance

Reply:
understood, let me think in a better way to perform this